src/systems/filecoin_blockchain/storage_power_consensus/_index.md

```diff
@@ -30,26 +30,114 @@ We must distinguish between both types of "miners" (storage and block miners). {
 
-## Repeated leader election attempts
+Because a Tipset can contain multiple blocks, the smallest ticket in the Tipset must be drawn otherwise the block will be invalid.
+
+```
+   ┌──────────────────────┐                     
+   │                      │                     
+   │                      │                     
+   │┌────┐                │                     
+   ││ TA │              A │                     
+   └┴────┴────────────────┘                     
+                                                
+   ┌ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─                      
+                          │                     
+   │                                            
+    ┌────┐                │       TA < TB < TC  
+   ││ TB │              B                       
+    ┴────┘─ ─ ─ ─ ─ ─ ─ ─ ┘                     
+                                                
+   ┌ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─                      
+                          │                     
+   │                                            
+    ┌────┐                │                     
+   ││ TC │              C                       
+    ┴────┘─ ─ ─ ─ ─ ─ ─ ─ ┘                     
+```
+
+In the above diagram, a miner will use block A's Ticket to generate a new ticket (or an election proof farther in the future) since it is the smallest in the Tipset.
+
+## Drawing randomness for sector commitments
+
+Tickets are used as input to the SEAL above in order to tie Proofs-of-Replication to a given chain, thereby preventing long-range attacks (from another miner in the future trying to reuse SEALs).
+
+The ticket has to be drawn from a finalized block in order to prevent the miner from potential losing storage (in case of a chain reorg) even though their storage is intact.
+
+Verification should ensure that the ticket was drawn no farther back than necessary by the miner. We note that tickets can uniquely be associated to a given round in the protocol (lest a hash collision be found), but that the round number is explicited by the miner in `commitSector`.
+
+We present precisely how ticket selection and verification should work. In the below, we use the following notation:
+
+- `F`-- Finality (number of rounds)
+- `X`-- round in which SEALing starts 
+- `Z`-- round in which the SEAL appears (in a block)
+- `Y`-- round announced in the SEAL `commitSector` (should be X, but a miner could use any Y <= X), denoted by the ticket selection
+ - `T`-- estimated time for SEAL, dependent on sector size
+ - `G = T + variance`-- necessary flexibility to account for network delay and SEAL-time variance. 
+
+We expect Filecoin will be able to produce estimates for sector commitment time based on sector sizes, e.g.:
+`(estimate, variance) <--- SEALTime(sectors)`
+G and T will be selected using these.
+
+#### Picking a Ticket to Seal
+
+When starting to prepare a SEAL in round X, the miner should draw a ticket from X-F with which to compute the SEAL.
```

Comment:
why is this here? shouldn't we just say tickets are how we produce randomness in the chain at a given epoch, and then let other things (ie seal) say how they use that randomness?